src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -840,16 +841,15 @@ class DrawSketchHandlerGenConstraint: public DrawSketchHandler
 
     virtual void activated(ViewProviderSketch *)
     {
-        Gui::Selection().rmvSelectionGate();
+        ongoingSequences = new std::set<int>(); //TODO: make it contain more values
+        _tempOnSequences = new std::set<int>();
+        selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
 
-        // Estimate allowed selections from the first types in allowedSelTypes
-        int allowedSelTypes = 0;
-        for (std::vector< std::vector< SelType > >::const_iterator it = cmd->allowedSelSequences.begin();
-             it != cmd->allowedSelSequences.end(); ++it) {
-            allowedSelTypes = allowedSelTypes | *((*it).begin());
-        }
-        GenericConstraintSelection* selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
-        selFilterGate->setAllowedSelTypes(allowedSelTypes);
+        resetOngoingSequences();
+
+        selSeq = *(new std::vector<SelIdPair>());
```

Comment:
I'm sorry I can't see how this creates a memory leak. Could you elaborate? Everytime a command is called, a new DrawSketchHandlerGenConstraint is created, and "selSeq = *(new std::vector<SelIdPair>());" is intended to be run only the first time. Before this point selseq doesn't exist so "selSeq.clear();" will fail. Perhaps I should move it to the constructor, along with the other constructions?